babel/messages/pofile.py

```diff
@@ -276,6 +279,17 @@ def _process_keyword_line(self, lineno, line, obsolete=False) -> None:
 
         self.obsolete = obsolete
 
+        def _normalized_string(s):
+            # whole lines are already stripped on both ends
+            s = s.lstrip()
+            if s[0] != '"':
+                self._invalid_pofile(line, lineno, "String must be delimited on the left by double quotes")
+                s = '"' + s
+            if s[-1] != '"':
```

Comment:
These would raise an ugly `IndexError` if `s` is falsy – can we be sure that'll never be the case?

Reply:
I thought we could.

- Each line is first stripped
- Keyword lines that trigger this code have a keyword followed by a `[` or by.a space (which implies at least another character after the space, because of the previous step)
- The incoming `s` here is `arg` in the enclosing context, i.e. what comes after the keyword, so either at least a `[` or at least a space and another character; hence, never empty after lstrip
  - The only problem here is that we could eventually be handling a line literally looking like `msgstr[1]` (or similarly invalid lines with no correct content after `]`) and we would run into the issue you mention. But OTOH it looks like the parsing of msgstr plural entries is currently quite fragile anyway (e.g., if no `]` is present, or what's between is not a correct number, or nothing in between, etc...) you would get other sorts of errors.

So: would you rather keep this change as-is and deal with msgstr plurals separately to add more validations, or should I go with a slice syntax to avoid any potential index errors?

Reply:
OK, I've opted for adding further checks to msgstr plural entries, ensuring that by the point the message itself is being parsed we can be sure it's not empty. I've also added an `assert` just before accessing the `0` and `-1` elements to denote that the string is expected to be not empty at that point.